File: UCFPT/211120/solutions/b/digits.java
/*
  digits.java
  Digit Solitaire, MCPC 2012 Problem B
  Java solution by Michael Goldwasser
*/

import java.util.*;
import java.io.*;

class digits {

    public static void main(String[] args) throws Exception {
        Scanner in = new Scanner(new File("digits.in"));

        while (true) {
            int v  = in.nextInt();
            if (v == 0) break;
            System.out.print(v);                   // initial number; no space
            while (v > 9) {
                // compute the product
                int product = 1;
                while (v > 0) {
                    product *= v % 10;
                    v /= 10;
                }
                System.out.print(" " + product);   // note leading space
                v = product;
            }
            System.out.println();                  // newline character
        }
    }
}

File: UCFPT/211120/solutions/b/digits.cpp
/*
  digits.cpp
  Digit Solitaire, MCPC 2012 Problem B
  C++ solution by Michael Goldwasser
*/

#include <iostream>
#include <fstream>
using namespace std;

ifstream fin("digits.in");

int main() {
    while (true) {
        int v;
        fin >> v;
        if (v == 0) break;
        cout << v;                     // initial number; no space
        while (v > 9) {
            // compute the product
            int product(1);
            while (v > 0) {
                product *= v % 10;
                v /= 10;
            }
            cout << " " << product;    // note leading space
            v = product;
        }
        cout << endl;
    }
}
